File: 2022_H1/OOPTADS_OOTISP/labs_ideaproject/src/main/java/com/notalonelyday/labs/lab3/jframe/Surface.java
package com.notalonelyday.labs.lab3.jframe;


import com.notalonelyday.labs.lab3.shape.round.Ellipse;
import com.notalonelyday.labs.lab3.shape.polygon.Rectangle;
import com.notalonelyday.labs.lab3.shape.polygon.Square;

import javax.swing.*;
import java.awt.*;

public class Surface extends JPanel {

    private void doDrawing(Graphics graphics) {

        Graphics2D g2d = (Graphics2D) graphics;
        g2d.setPaint(new Color(120, 120, 120));

        RenderingHints rh = new RenderingHints(
                RenderingHints.KEY_ANTIALIASING,
                RenderingHints.VALUE_ANTIALIAS_ON);
        rh.put(RenderingHints.KEY_RENDERING,
                RenderingHints.VALUE_RENDER_QUALITY);

        g2d.setRenderingHints(rh);
    }

    @Override
    public void paintComponent(Graphics g) {
        super.paintComponent(g);
        doDrawing(g);
        drawFigures(g);
    }

    public void drawFigures(Graphics graphics) {
//        DrawerKt.drawShape(graphics, new StraightLine(
//                10, 10, 10, 100
//        ), 1);

        DrawerKt.drawShape(
                graphics,
                new Rectangle(
                        10,10,100,200
                ), 1
        );
        DrawerKt.drawShape(
                graphics,
                new Square(
                        100,100,100
                ), 1
        );
        DrawerKt.drawShape(
                graphics,
                new Ellipse(
                        500,500,400, 400
                ), 1
        );
    }
}

File: 2022_H1/OOPTADS_OOTISP/labs_ideaproject/src/main/java/com/notalonelyday/labs/lab3/jframe/Surface.kt
package com.notalonelyday.labs.lab3.jframe

import com.notalonelyday.labs.lab3.shape.base.AbstractShape
import com.notalonelyday.labs.lab3.shape.line.StraightLine
import java.awt.*
import java.awt.event.MouseEvent
import java.awt.event.MouseListener
import javax.swing.BorderFactory
import javax.swing.JPanel
import javax.swing.border.Border

object Surface : JPanel(), MouseListener {

    init {
        addMouseListener(this)
        isVisible = true
        border = BorderFactory.createMatteBorder(4, 4, 4, 4, Color.RED)
    }
    private fun doDrawing(graphics: Graphics) {
        val g2d = graphics as Graphics2D
        g2d.paint = Color(120, 120, 120)
        val rh = RenderingHints(
            RenderingHints.KEY_ANTIALIASING,
            RenderingHints.VALUE_ANTIALIAS_ON)
        rh[RenderingHints.KEY_RENDERING] = RenderingHints.VALUE_RENDER_QUALITY
        g2d.setRenderingHints(rh)
    }

    var g: Graphics? = null
    val shapes = mutableListOf<AbstractShape>()
    public override fun paintComponent(g: Graphics) {
        super.paintComponent(g)
        doDrawing(g)
        this.g = g
        shapes.forEach { g.drawShape(it) }
    }

    fun drawShape(shape: AbstractShape) {
        shapes.add(shape)
        repaint()
    }

    val points = mutableListOf<Point>()

    private fun addPoint(p: Point){
        println(p)
        if (points.lastOrNull() != p) {
            points.add(0, p)
            refreshPoints()
            Controls.addPoint(p)
        }
    }

    private fun refreshPoints(){

    }

    override fun mouseClicked(mouseEvent: MouseEvent) {
        addPoint(mouseEvent.point)
    }
    override fun mousePressed(mouseEvent: MouseEvent) {
    }
    override fun mouseReleased(mouseEvent: MouseEvent) {
    }
    override fun mouseEntered(mouseEvent: MouseEvent) {
    }
    override fun mouseExited(mouseEvent: MouseEvent) {
    }
}